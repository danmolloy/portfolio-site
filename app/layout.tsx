import { Metadata } from 'next'
import Header from './layout/header'
import Footer from './layout/footer'
import "./globals.css";
import { Inconsolata, Raleway, Poiret_One, Poppins, Rubik, Noto_Serif_JP, Crimson_Text, PT_Serif, Quicksand } from 'next/font/google'
import { Analytics } from "@vercel/analytics/react";

const quicksand = Quicksand({weight: "400", subsets: ['latin'], variable: "--font-quicksand"})
//const poiretOne = Poiret_One({weight: '400', subsets: ["latin"], variable:"--font-poiret"})
const notoSerif = PT_Serif({weight: '400',subsets: ["latin"], variable:"--font-noto-serif"})

 
export const metadata: Metadata = {
  title: 'Daniel Molloy | Developer',
  description: 'Dan Molloy is an Australian web developer based in London.',
}

export default function RootLayout({
  // Layouts must accept a children prop.
  // This will be populated with nested layouts or pages
  children,
}: {
  children: React.ReactNode
}) {
  return (
    <html lang="en" className={`${quicksand.variable} ${notoSerif.variable}`}>
      <body>
      
    <div data-testid="layout" className={` w-full min-h-screen bg-white flex flex-col items-center justify-between font-sans dark:bg-slate-900 dark:text-blue-300`}>
      <Header />
      <main className={` flex flex-col items-center`}>
      {children}
      </main>
      <Footer />
    </div>
  
      </body>
      <Analytics />
    </html>
  )
}