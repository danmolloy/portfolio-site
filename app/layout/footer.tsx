'use client'
import Link from "next/link"
import React from "react"


export default function Footer() {
  return (
    <div data-testid="footer" className="w-full z-10 h-20  flex sm:flex-row flex-col sm:justify-between p-2 items-center border-t  dark:border-t-zinc-600">
      <Link href="/" className=' hover:underline text-sm font-light hover:cursor-pointer active:text-zinc-400 m-1 ' >
        Back to top
      </Link>
      <p className="text-sm  text-zinc-800 dark:text-zinc-400">Designed and built by Daniel Molloy</p>
    </div>
  )
}