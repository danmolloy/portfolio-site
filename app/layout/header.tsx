import { BsMoon } from "react-icons/bs"

import ToggleSwitch from './toggle'
import Link from "next/link"


export default function Header() {

  return (
    <div
      data-testid="header"
    className={" dark:bg-gray-800 justify-between  transition duration-500 ease-out fixed  bg-white z-30 h-14 w-screen top-0 flex flex-row  items-center  shadow fill-zinc-300 text-black"}>
        <Link href={"/"} data-testid="home-btn" className='font-sans p-4 m-2 text-lg active:text-zinc-400 dark:text-blue-400 dark:hover:text-yellow-300' >
          Daniel Molloy
        </Link>
      
      <div className=' flex flex-row items-center justify-center mx-4'>
      <ToggleSwitch />
        <div className="dark:text-yellow-400 text-slate-200 transition-colors duration-300 ">
          <BsMoon />
        </div>
        </div>
    </div>
  )
}