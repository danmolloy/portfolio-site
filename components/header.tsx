import Link from "next/link"
import React from "react"
import { BsSun, BsMoon } from "react-icons/bs"

export default function Header() {

  const handleToggle = () => {
    if (!localStorage.theme){
      document.documentElement.classList.add('dark')
      localStorage.setItem("theme", "dark")
    } else {
      document.documentElement.classList.remove('dark')
      localStorage.removeItem("theme")
    }
    
  }

  return (
    <div className="dark:bg-zinc-900 flex flex-row justify-between w-full h-16 border-b dark:border-b-zinc-700 items-center px-4">
      <Link href="/">
        <h1 className="text-emerald-500 dark:hover:bg-zinc-800 hover:bg-emerald-50 p-1 rounded">DM</h1>
      </Link>
      <button onClick={() => handleToggle()}>
      <div className="dark:flex hidden text-emerald-500 hover:text-emerald-400 p-2 rounded-full">
          <BsSun />
        </div>
        <div className="dark:hidden text-emerald-500 hover:text-emerald-400 p-2 rounded-full">
          <BsMoon />
        </div>
      </button>
    </div>
  )
}