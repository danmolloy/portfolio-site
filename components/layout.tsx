import React from "react";
import Footer from "./footer";
import Header from "./header";

type LayoutProps = {
  children: React.ReactNode
}

export default function Layout({ children }: LayoutProps) {
  return (
    <div className="w-full min-h-screen bg-white flex flex-col justify-between">
      <Header />
      <main>
      {children}
      </main>
      <Footer />
    </div>
  )
}