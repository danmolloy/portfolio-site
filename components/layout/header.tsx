import { useEffect, useState } from 'react'
import { AiOutlineMenu } from 'react-icons/ai'
import { BsSun, BsMoon } from "react-icons/bs"

import MenuIcon from '../menuIcon'
import { Link, animateScroll as scroll } from 'react-scroll'
import ToggleSwitch from './toggle'

export const menuItems: {title: string}[] = [
  {
    title: "About",
  },
  {
    title: "Projects",
  },
  {
    title: "Contact",
  },
]

export type HeaderProps = {
  showMenu: boolean
  setShowMenu: () => void
  setShowHeader: (arg: boolean) => void
  showHeader: boolean
}

export default function Header(props: HeaderProps) {
  const [scrollPosition, setScrollPosition] = useState(0);
  const { showMenu, setShowMenu, setShowHeader, showHeader } = props;
  //const [showHeader, setShowHeader] = useState(true)
  const [darkMode, setDarkMode] = useState<boolean>(false)

  useEffect(() => {
    if (localStorage.theme === 'dark' || (!('theme' in localStorage) && window.matchMedia('(prefers-color-scheme: dark)').matches)) {
      document.documentElement.classList.add('dark')
      setDarkMode(true)
    } else {
      document.documentElement.classList.remove('dark')
      localStorage.removeItem("theme")
      setDarkMode(false)
    }
  }, [])

  const handleToggle = () => {
    if (!localStorage.theme || darkMode === false){
      document.documentElement.classList.add('dark')
      localStorage.setItem("theme", "dark")
      setDarkMode(true)
    } else {
      document.documentElement.classList.remove('dark')
      localStorage.removeItem("theme")
      setDarkMode(false)
    }
    
  }

  useEffect(() => {
    function handleScroll() {
      const currentPosition = window.scrollY;
      if (currentPosition > scrollPosition + 25) {
        setScrollPosition(currentPosition);
        setShowHeader(false)
      } else if (currentPosition < scrollPosition - 50) {
        setScrollPosition(currentPosition);
        setShowHeader(true)
      }
    }

    window.addEventListener('scroll', handleScroll);

    return () => {
      window.removeEventListener('scroll', handleScroll);
    };
  }, [scrollPosition]);


  return (
    <div
      data-testid="header"
    className={!showHeader && scrollPosition < 750
      ? "dark:bg-gray-800 justify-between font-display -translate-y-full transition duration-1000  fixed z-30 h-14 w-screen top-0 flex flex-row-reverse md:flex-row items-center fill-zinc-300 text-zinc-100"  
      : !showHeader
      ? "dark:bg-gray-800 justify-between font-display -translate-y-full transition duration-500 bg-white fixed z-30 h-14 w-screen top-0 flex flex-row-reverse md:flex-row items-center  shadow fill-zinc-300 text-black"  
      : "dark:bg-gray-800 justify-between font-display transition duration-500 ease-out fixed  bg-white z-30 h-14 w-screen top-0 flex flex-row-reverse md:flex-row  items-center  shadow fill-zinc-300 text-black"}>
        <div className='hidden md:flex w-screen  flex-row  justify-end pr-4'>
        <button data-testid="home-btn" className=' p-4 m-2 text-lg font-light hover:cursor-pointer active:text-zinc-400 dark:text-blue-400 dark:hover:text-yellow-300' onClick={() => scroll.scrollToTop({ smooth: true, duration: 500 })}>
          Home
        </button>
      {menuItems.map(i => (
        <Link onClick={() => setTimeout(() => setShowHeader(false), 750)} activeClass="active" to={i.title.toLowerCase()} spy={true} smooth={true} offset={50} duration={500}  key={i.title} className='dark:text-blue-400 dark:hover:text-yellow-300 p-4 m-2 text-lg font-light hover:cursor-pointer active:text-zinc-400'>
          <p>{i.title}</p>
        </Link>
      ))}
      </div>
      <div className=' flex flex-row items-center justify-center mx-4'>
      <ToggleSwitch setToggled={() => handleToggle()} toggled={darkMode} label='' />
        <div className="dark:text-yellow-400 text-slate-200 transition-colors duration-300 ">
          <BsMoon />
        </div>
        </div>
      <MenuIcon setShowMenu={() => setShowMenu()} showMenu={showMenu}/>
    </div>
  )
}