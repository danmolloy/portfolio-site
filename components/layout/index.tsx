import React, { useState } from "react";
import Footer from "./footer";
import Header from "./header";
import Menu from "../menu";

export type LayoutProps = {
  children: React.ReactNode
}

export default function Layout({ children }: LayoutProps) {
  const [showMenu, setShowMenu] = useState(false)
  const [showHeader, setShowHeader] = useState(true)
  
  return (
    <div data-testid="layout" className="dark:bg-zinc-900 dark:text-zinc-300 w-full min-h-screen bg-white flex flex-col items-center justify-between font-sans">
      <Header showHeader={showHeader} setShowHeader={(i) => setShowHeader(i)} setShowMenu={() => setShowMenu(!showMenu)} showMenu={showMenu} />
      <Menu showMenu={showMenu} setShowHeader={i => setShowHeader(i)} setShowMenu={i => setShowMenu(i)} />
      <main className={showMenu ? "transition duration-500 blur-sm flex flex-col items-center" : "transition duration-500 flex flex-col items-center"}>
      {children}
      </main>
      <Footer />
    </div>
  )
}