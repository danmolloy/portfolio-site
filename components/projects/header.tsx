import { FaLaptopCode, FaLink } from "react-icons/fa"

export type ProjectHeaderProps = {
  title: string
  githubLink: string
  previewLink: string
  selectedTab: number
  setSelectedTab: (arg: number) => void
}

export default function ProjectHeader(props: ProjectHeaderProps) {
  const { githubLink, previewLink, selectedTab, setSelectedTab, title } = props;

  return (
    <div data-testid="project-header" className=" w-full">
      <div className="flex flex-row justify-between items-start w-full">
      <h2 className="text-blue-700 dark:text-blue-400">{title}</h2>
      <div className="flex flex-col">
        <a data-testid="github-link" href={githubLink} target="_blank" className="m-1 ml-2  flex flex-row items-center text-sm hover:text-blue-500 dark:hover:text-yellow-300">
          <FaLaptopCode />
          <p className="ml-1 hover:text-black dark:hover:text-yellow-300">
            View GitHub
          </p>
        </a>
        <a data-testid="example-link" href={previewLink} target="_blank" className="m-1 flex flex-row items-center text-sm hover:text-blue-500 dark:hover:text-yellow-300">
          <FaLink />
          <p className="ml-1 hover:text-black dark:hover:text-yellow-300">
            Live Example
          </p>
        </a>
        </div>
        </div>
      <div className="w-full flex flex-row mb-6">
        <button data-testid="overview-btn" className={selectedTab === 0 ? "custom-border-gradient border-b border-blue-500  py-2 my-2 mx-4 hover:text-zinc-400 " : "py-2 my-2 mx-4 hover:text-zinc-400"} onClick={() => setSelectedTab(0)}>Overview</button>
        <button className={selectedTab === 1 ? "custom-border-gradient border-b border-blue-500  py-2 my-2 mx-4 hover:text-zinc-400 " : "py-2 my-2 mx-4 hover:text-zinc-400" } onClick={() => setSelectedTab(1)}>Features</button>
      </div>
    </div>
  )
}