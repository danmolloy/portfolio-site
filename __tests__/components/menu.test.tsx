import { menuItems } from "@/components/layout/header";
import Menu, { MenuProps } from "@/components/menu";
import { act, fireEvent, render, screen } from "@testing-library/react";
import { Link, animateScroll } from 'react-scroll'

jest.mock("react-scroll", () => ({
  Link: (props: any) => <div />,
  animateScroll: {
    scrollToTop: jest.fn()
  }
}))

const mockProps: MenuProps = {
  setShowMenu: jest.fn(),
  setShowHeader: jest.fn(),
  showMenu: false
}

describe("<Menu />", () => {
  beforeEach(() => {
    render(<Menu {...mockProps} />)
  })
  it("menu is in the document", () => {
    const menu = screen.getByTestId("menu")
    expect(menu).toBeInTheDocument()
  })
  it("Home btn is in the document and calls scrollToTop on click", () => {
    const homeBtn = screen.getByText("Home")
    expect(homeBtn).toBeInTheDocument()
    act(() => {
      fireEvent.click(homeBtn)
    })
    expect(mockProps.setShowMenu).toHaveBeenCalledWith(false)
    expect(animateScroll.scrollToTop).toHaveBeenCalledWith({smooth: "true", duration: 500})
  })
  it("matches snapshot", () => {
    const menu = screen.getByTestId("menu")
    expect(menu).toMatchSnapshot()
  })
})