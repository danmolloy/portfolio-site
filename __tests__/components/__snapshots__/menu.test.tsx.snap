// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Menu /> matches snapshot 1`] = `
<div
  class=" ease-in -translate-y-full transition duration-500 opacity-90 rounded-bl flex flex-col z-20 bg-white fixed w-screen  min-h-screen right-0 shadow text-2xl"
  data-testid="menu"
>
  <div
    class="flex flex-col w-full justify-evenly  font-display mt-16"
  >
    <button
      class="py-4 m-4 flex flex-col text-black  items-center justify-center hover:text-orange-500 active:text-amber-400"
    >
      Home
    </button>
    <div />
    <div />
    <div />
  </div>
</div>
`;
